operator/internal/controller/redpanda/redpanda_controller_test.go

```diff
@@ -398,6 +398,24 @@ func (s *RedpandaControllerSuite) TestClusterSettings() {
 		// Only assert that c.Expected is a subset of the set config.
 		// The chart/operator injects a bunch of "useful" values by default.
 		s.Subset(config, c.Expected)
+
+		// check that the pods are annotated with the config version
```

Comment:
I think Rafal mentioned this as well, but it would be good to:
- include a config change that requires a restart.
- Ping the admin API to assert that redpanda doesn't report a need for a restart.